new-log-viewer/src/components/Layout.tsx

```diff
@@ -16,148 +44,195 @@ import {
-        name: LOCAL_STORAGE_KEY.DECODER_OPTIONS_TIMESTAMP_KEY,
-        type: "text",
-    },
-    {
-        initialValue: getConfig(CONFIG_KEY.THEME),
-        label: LOCAL_STORAGE_KEY.THEME,
-        name: LOCAL_STORAGE_KEY.THEME,
-        type: "text",
-    },
-    {
-        initialValue: getConfig(CONFIG_KEY.PAGE_SIZE),
-        label: LOCAL_STORAGE_KEY.PAGE_SIZE,
-        name: LOCAL_STORAGE_KEY.PAGE_SIZE,
-        type: "number",
-    },
-];
 
 /**
- * Renders a form for testing config utilities.
  *
- * @return
+ * @param actionName
+ * @param logEventNum
+ * @param numEvents
  */
-const ConfigForm = () => {
-    const handleConfigFormReset = (ev: React.FormEvent) => {
-        ev.preventDefault();
-        window.localStorage.clear();
-        window.location.reload();
-    };
-
-    const handleConfigFormSubmit = (ev: React.FormEvent) => {
-        ev.preventDefault();
-        const formData = new FormData(ev.target as HTMLFormElement);
-
-        const formatString = formData.get(LOCAL_STORAGE_KEY.DECODER_OPTIONS_FORMAT_STRING);
-        const logLevelKey = formData.get(LOCAL_STORAGE_KEY.DECODER_OPTIONS_LOG_LEVEL_KEY);
-        const timestampKey = formData.get(LOCAL_STORAGE_KEY.DECODER_OPTIONS_TIMESTAMP_KEY);
-        const theme = formData.get(LOCAL_STORAGE_KEY.THEME);
-        const pageSize = formData.get(LOCAL_STORAGE_KEY.PAGE_SIZE);
-        let error = null;
-        if (
-            "string" === typeof formatString &&
-            "string" === typeof logLevelKey &&
-            "string" === typeof timestampKey
-        ) {
-            error ||= setConfig({
-                key: CONFIG_KEY.DECODER_OPTIONS,
-                value: {formatString, logLevelKey, timestampKey},
+const handleAction = (actionName: ACTION_NAME, logEventNum: number, numEvents: number) => {
+    const pageSize = getConfig(CONFIG_KEY.PAGE_SIZE);
+    switch (actionName) {
+        case ACTION_NAME.FIRST_PAGE:
+            updateWindowUrlHashParams({logEventNum: 1});
+            break;
+        case ACTION_NAME.PREV_PAGE:
+            updateWindowUrlHashParams({
+                logEventNum: getLastItemNumInPrevChunk(logEventNum, pageSize),
             });
-        }
-        if ("string" === typeof theme) {
-            error ||= setConfig({
-                key: CONFIG_KEY.THEME,
-                value: theme as THEME_NAME,
-            });
-        }
-        if ("string" === typeof pageSize) {
-            error ||= setConfig({
-                key: CONFIG_KEY.PAGE_SIZE,
-                value: Number(pageSize),
+            break;
+        case ACTION_NAME.NEXT_PAGE:
+            updateWindowUrlHashParams({
+                logEventNum: getFirstItemNumInNextChunk(logEventNum, pageSize),
             });
+            break;
+        case ACTION_NAME.LAST_PAGE:
+            updateWindowUrlHashParams({logEventNum: numEvents});
+            break;
+        default:
+            break;
+    }
+};
+
+/**
+ *
+ */
+const MenuBar = () => {
```

Comment:
Shall we move this into another source file?

Reply:
Sure. Will do